### ISSUE TITLE
P3P refuses to start upgrade process after forced downgrades

### ISSUE THREAD

**Issue by DannyBoyCJ** [mefistotelis/phantom-firmware-tools#37]
Hello guys, I am new to this so go easy on me please...  I upgraded my P3P to version P3X_FW_V01.10.0090, following the DJI guide, and got a failed upgrade signal and the dreaded message in the log file:

[15 00][00] device not detected
[08 00][00] device not detected

After this, there was no video transmission. Then I googled the issue and found a guy who said he resolved this by downgrading all the way to the first version available to him (by the "debug" file method), and then upgrading again, step by step, up to version P3X_FW_V01.08.0080, at which time he got a successful upgrade process, and a functional drone. I proceeded to do the same, but when I tried to go from 1.2.3 to 1.1.9, it went wrong. The process didn't finish, and I found the drone with a limp gimbal, the camera just hanging at an angle underneath, beeping. The last thing written in the log file is: 

 [00055385]Firmware upgrading[1]...
 [00066342][01 00] Firmware upgrade start...

My problem is that now I can't get it to enter the upgrade mode no matter what firmware version  I place on the card, with or without the "Debug" file. When I try to force start the upgrade process by pressing the link button during startup, the gimbal immediately goes limp and starts beeping (but not the normal beeps it emits during upgrade). It actually resumes the state in which I found it after the last failed upgrade. It does nothing and writes no log, no matter how long I leave it at it.

This is the last log file, written at the time of that last failed downgrade:
[P3X_FW_LOG_AB.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1118079/P3X_FW_LOG_AB.txt)

So, if you have any ideas, or know of a procedure to make it upgrade again (via serial service pads, or any other method), please help.  Also, if you need more info, just let me know.
Thank you!

**Comment by mefistotelis** [comment]
Connect to Ambarella serial console (U0 pins) and check what you see.
If there's nothing - you have to reflash MX30LF by hand. If there's something - maybe you will be able to get into bootloader and flash from sd-card.

**Comment by DannyBoyCJ** [comment]
Thank you, I will try this and post back. Just to be clear, do I need to power the board outside the drone to have access to the pins, or do I do this with the drone (and gimbal) assembled normally, and just run thin wires from the U0 pins to the TTL-USB converter?  If necessary, I have a stabilized variable voltage source that I can use. I just want to make sure I understand the procedure correctly. Also, if there's a thread on this that you know of, please share the link... Thank you!

**Comment by mefistotelis** [comment]
>  do I need to power the board outside the drone to have access to the pins, or do I do this with the drone (and gimbal) assembled normally

Whichever works best for you. You will not need communication with any other part of the drone to flash Ambarella FW.

>  if there's a thread on this that you know of, please share the link

There is. [Link](https://www.google.pl/search?q=phantom+gimbal+%22u0_tx%22+%22u0_rx%22).

**Comment by DannyBoyCJ** [comment]
Thanks a lot, I'll try this tomorrow and get back with the results!

**Comment by DannyBoyCJ** [comment]
Ok, so it looks alive... This is the log from Putty:

[Putty_log_1_U0_R_U0_T.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1122646/Putty_log_1_U0_R_U0_T.txt)

How do you think I should proceed based on this?

**Comment by notsolowki** [comment]
Looks like the camera is bricked, it dont detect the sdcard

**Comment by DannyBoyCJ** [comment]
The SD card wasn't inside, I'll try to do it again with it mounted and post back...

**Comment by DannyBoyCJ** [comment]
Ok, this is weird, I think it just started updating from SD (I had firmware on it), still connected to the TTL adapter. This is what it's done so far:

[Putty_log_2_U0_R_U0_T.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1122736/Putty_log_2_U0_R_U0_T.txt)

Do I let it go on? Seems like it's in a loop to me.

**Comment by DannyBoyCJ** [comment]
Ok, definitely a loop, I stopped it. Bad blocks... is that solvable?

**Comment by mefistotelis** [comment]
You'd probably have to replace the NAND chip. I don't have experience in it.
@coptersafe, any advice for NAND with bad blocks?

**Comment by MrBurnsAT** [comment]
Should Repair itself by powering on Time by Time.
Same Problem here with NAND run Full of Bad Blocks at an HDMI Output Module

**Comment by DannyBoyCJ** [comment]
Does that mean powering it on and off repeatedly? Is that with or without the firmware on the SD? And for how long should I let it run each time?

**Comment by notsolowki** [comment]
Maybe the chip itself became unseated from The pcb

**Comment by DannyBoyCJ** [comment]
The chip looks completely flush with the board. Of course, that's no guarantee... but the drone never had a physical shock. And it all happened while it was sitting on my desk, updating. I could try to do a reflow (I've done it before on a laptop video-board chip). I am bit reluctant though, because there are a ton of very small components around this chip, and my heating equipment isn't exactly precise. Just to be clear, the nand chip is this one...

![mxic](https://user-images.githubusercontent.com/25065355/27858070-832274b0-617c-11e7-89a2-a6d685d12680.jpg)

...right? The MXIC. I wouldn't want to work on the wrong one. In the meantime, does anyone have a spare gimbal board like this? I might have to buy another one.

**Comment by DannyBoyCJ** [comment]
This is what I get when I connect to 368_U0 pins:

DM36x initialization passed!
UBL Product Vesion : DJI-NOHDMI-WM610-UBL-1.0-rc0(2015-01-13)
Dji UBL Version: 1.51(Jan 13 2015 - 16:42:11)
Booting Catalog Boot Loader
BootMode = NAND
Starting NAND Copy...
Valid magicnum, 0xA1ACED66, found in block 0x00000019.
Valid magicnum, 0xA1ACED66, found in block 0x0000001D.
No valid boot image found!
NAND Boot failed.
Aborting...

Is there a way to boot from another device instead of the nand memory?

**Comment by mefistotelis** [comment]
> Is there a way to boot from another device instead of the nand memory?

Don't know.
But there is a way to update firmware via that serial connection:
https://github.com/mefistotelis/phantom-firmware-tools/wiki/Firmware-m0800#by-service-serial-port

**Comment by MrBurnsAT** [comment]
Try flashing my Package.

[HDMI.Final.zip](https://github.com/mefistotelis/phantom-firmware-tools/files/1129375/HDMI.Final.zip)

With Tools from here
[LB_hack.zip](https://github.com/mefistotelis/phantom-firmware-tools/files/1129377/LB_hack.zip)

**Comment by DannyBoyCJ** [comment]
Thanks guys, I'll try anything at this point.... I'll post back the results.

**Comment by MrBurnsAT** [comment]
here are the right ubl1 and uboot images for ur board!!!

i decrypte the firmware file and edited the ubl1 and uboot file that u can flash it with the tool from LB Hack


[Uboot OFMD.zip](https://github.com/mefistotelis/phantom-firmware-tools/files/1130599/Uboot.OFMD.zip)

**Comment by DannyBoyCJ** [comment]
Thanks a lot, I'll give it a try!

**Comment by DannyBoyCJ** [comment]
One more question before I go at it: which one is the service pad for boot select on the p3p top gimbal board? I found BOOT0 ans B_S as possible candidates so far...

**Comment by DannyBoyCJ** [comment]
I tried shorting BOOT0 to 3.3VD, but it doesn't say BOOTME. It tries to boot normally, from NAND:

DM36x initialization passed!
UBL Product Vesion : DJI-NOHDMI-WM610-UBL-1.0-rc0(2015-01-13)
Dji UBL Version: 1.51(Jan 13 2015 - 16:42:11)
Booting Catalog Boot Loader
BootMode = NAND
Starting NAND Copy...
Valid magicnum, 0xA1ACED66, found in block 0x00000019.
Valid magicnum, 0xA1ACED66, found in block 0x0000001D.
No valid boot image found!
NAND Boot failed.
Aborting...

 The same with B_S pad shorted to 3.3VD. Am I missing something?

**Comment by MrBurnsAT** [comment]
Try Shorting BOOT0 with VCC3V3

**Comment by MrBurnsAT** [comment]
Found this in ner

Try Shorting BT01, BT02 and 3.3

**Comment by MrBurnsAT** [comment]
![screenshot_20170707-152046](https://user-images.githubusercontent.com/24981264/27959759-fcbdc858-6328-11e7-912e-176a20b116e7.png)

![58fef174dc0a2a087cbe2d3b](https://user-images.githubusercontent.com/24981264/27959790-1b9f5908-6329-11e7-992e-e6d0bb1420d6.jpg)

**Comment by DannyBoyCJ** [comment]
Excellent advice, MrBurnsAT! It worked. I had to Ctr+C from the "BOOTME BOOTME" loop two times (and repeat the "-nandflash" command), but it finally took, and ended in "operation successfull". I guess I'm going to try an old fashion SD card FW upgrade next, I'm just not sure what version I should use. The last one I flashed was 1.1.9 (coming from 1.2.3, with debug), but it never finished upgrading. If you have any advice at this stage, it would be very welcome.

**Comment by mefistotelis** [comment]
Good. Updated wiki with your findings.

**Comment by MrBurnsAT** [comment]
Yesssssss

One with an Working Unit!
Im struggling with my HDMI Board which Think ist an Lightbridge 😂😂😂

**Comment by DannyBoyCJ** [comment]
Oh man, your HDMI Board who thinks it's a Lightbridge is awesome! I read your thread and it's gold. And yes, thanks a lot, I do have a working(ish) unit now! I'm actually getting image on my tablet (albeit delayed and at low frame rate), without doing any  firmware upgrading. I will have to, though, because the app says there's a mismatch in the firmware. Also, my remote is now on a newer version, so I might have to downgrade that too. I will start by trying to get a clean 1.2.3 version on the aircraft and, if successful, work my way up from there. Also, a very big thumbs up to mefistotelis, for creating and maintaining this outstanding knowledge pool!

**Comment by DannyBoyCJ** [comment]
Well, sadly, my drone still won't update. I still get this fom Ambarella:

[00057809][CA9_0] Program "System Software" to NAND flash ...
[00060922][CA9_0] Initial bad block. <block 173>
[00060967][CA9_0] Done
[00061171][CA9_0] Program "DSP uCode" to NAND flash ...
[00062498][CA9_0] Done
[00063108][CA9_0] Program "System ROM Data" to NAND flash ...
[00063108][CA9_0] Initial bad block. <block 309>
[00063241][CA9_0] Initial bad block. <block 313>
[00067047][CA9_0] Done
[00067330][CA9_0] Program "Linux Kernel" to NAND flash ...
[00069145][CA9_0] Done
[00070592][CA9_0] Program "Linux Root FS" to NAND flash ...
[00071974][CA9_0] Initial bad block. <block 541>
[00072018][CA9_0] Initial bad block. <block 543>
[00079742][CA9_0] Initial bad block. <block 886>
[00079791][CA9_0] Done
[00080123][CA9_0] [DJI] Done!
[00080123][CA9_0] [DJI] Reboot...... 

I wonder if I should erase the NAND before flashing it... Would that mark the bad blocks?

**Comment by mefistotelis** [comment]
This log comes from Ambarella, not DM36x. It's an unrelated problem. Different NAND chip.

**Comment by DannyBoyCJ** [comment]
Sorry, my  bad, I thought they were using the same chip... so here's the log from DM36x, with the SD card inside (with firmware on it):
[putty_09_07_2017_025755.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1133457/putty_09_07_2017_025755.txt)
The errors at the end are probably there because the camera is not connected to the top gimbal board. I tried the procedure with the drone fully assembled, still won't update. And the gimbal still goes limp when I try to force-update by link button, or try to run the service file (MVOM0Fw).

**Comment by Mavic129** [comment]
this message means the DM36x is not connected to the LB
GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1
ARM Load: 7% Video fps: 8 fps Video bit rate: 0 kbps PHY Bandwidth: 0 kbps Time: 00:00:03 Resolution: 1280x720

**Comment by mefistotelis** [comment]
> this message means the DM36x is not connected to the LB
...
and from the log i can tell the DM36x is functioning properly

Fully agree with @Mavic129. As I previously stated, you have an issue with Ambarella. DM36x is fixed.

**Comment by DannyBoyCJ** [comment]
Ok, that is not good news. I don't know what to do next. From the Ambarella log, I take it that it is flashing it's own NAND. Maybe I should try flashing the LB? Is that possible on the P3P?

**Comment by MrBurnsAT** [comment]
Loading from nand0, offset 0x4a0000
** Unknown image type
Wrong Image Format for bootm command
ERROR: can't get kernel image!


Same Problem as at my HDMI Board.

Im Thinking now:
We are only flashing ubl1 and u-boot File.

But in decodet Firmware there is a uImage File too. 
Which contains the Linux of DM368 

But i dont know how to Flash it.

Look at the decrypted Firmware in
/dji/bin/loadmodule-rc
And 
/dji/etc/updatefinish

**Comment by MrBurnsAT** [comment]
thats in /dji/etc/updatefinish

```
#!/bin/sh
rm -f /opt/tmp/update.tar.gz

UBL1="/opt/dji/data/ubl1.img"
UBL2="/opt/dji/data/ubl2.img"
UBOOT="/opt/dji/data/u-boot.img"
KERNEL="/opt/dji/data/uImage"

RCS="/opt/dji/etc/rcS"

if [ -f $UBL1 ]
then
	echo "mtd write $UBL1"
	mtdwritefile -f $UBL1 -o 0x20000
	rm -rf $UBL1
fi

if [ -f $UBL2 ]
then
	echo "mtd write $UBL2"
	mtdwritefile -f $UBL2 -o 0x40000
	rm -rf $UBL2
fi

if [ -f $UBOOT ]
then
	echo "mtd write $UBOOT"
	mtdwritefile -f $UBOOT -o 0x320000
	rm -rf $UBOOT
	echo "delete uboot env"
	mtderase -o 0x420000 -l 0x40000
fi

if [ -f $KERNEL ]
then
	echo "mtd write $KERNEL"
	mtdwritefile -f $KERNEL -o 0x4a0000
	rm -rf $KERNEL
fi

if [ -f $RCS ]
then
	echo "replace rcS"
	mv $RCS /etc/init.d/
fi


sync
#reboot
```

**Comment by DannyBoyCJ** [comment]
Hmm... I see it now... Maybe it can be flashed by the same method. MrBurnsAT, do you perhaps have this image for Linux (or is there a way to extract it)? You say you have the same problem with your HDMI board, have you tried flashing the Linux image by any method, even if not successful, yet? Also, how do you access the Linux system on the drone? (/dji/etc/updatefinish). Sorry for asking such questions, but as I said, I have no experience and I'm learning as I go here...

**Comment by MrBurnsAT** [comment]
No Not in Linux System!!!

In the extracted Firmware
Decrypted.tar.gz

Into that File are all These folders

In Folder Data u have the ubl1, ubl2, u-boot and uImage Files

No i wasnt able to Flash the uImage File.

Because i dont know how

**Comment by DannyBoyCJ** [comment]
Oh, ok, thanks. I'll research this, there must be a way to do it. If I find anything useful, I'll post it here.

**Comment by MrBurnsAT** [comment]
If i find something Ill Post it too

**Comment by DannyBoyCJ** [comment]
Updating a kernel from a linux prompt
If you want to be able to update your kernel or uboot image in NAND (I'm using a DM355 platform, but this should work for others) *while* your system is running and without the use of a JTAG device, these are the commands that can be used to do this:

   /usr/sbin/flash_eraseall -j /dev/mtd2
   /usr/sbin/nandwrite -p /dev/mtd2 uImage

I found this method here:
http://processors.wiki.ti.com/index.php/Writing_Kernel_Image_to_NAND_Flash#Writing_the_Kernel_Image_into_NAND_Flash

The page is for DM355 and does not apply to DM36x, but, as written above, this particular way of flashing "should work for other" platforms. So... how do  we access shell on our chip? And do you think this would work?

Another possibly useful link: http://processors.wiki.ti.com/index.php/UG:_DaVinci_PSP_Installation_on_DM36x_EVM

**Comment by DannyBoyCJ** [comment]
Hey guys, apparently the GetFPGAstatus error was indeed caused by the camera not being attached to the main gimbal board. Here is the log with the camera attached (and no FPGA errors):

[putty_10_07_2017_012219.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1134132/putty_10_07_2017_012219.txt)

The link to LB works, but the drone still won't take an update from SD, though.
MrBurnsAT, regarding the kernel image, as you said, at first it reports:

--------------------------------
Loading from nand0, offset 0x4a0000
** Unknown image type
Wrong Image Format for bootm command
ERROR: can't get kernel image!
--------------------------------

I think the Linux image is not located at offset 0x4a0000. Only the "bootm" command is located there, and it is probably corrupted. But a duplicate of this command is located at offset 0x900000 (maybe a backup?), and it starts executing after the first one fails. The "Load Address" and name of the Linux image (80008000) is the same as the one in the sample here:
https://github.com/mefistotelis/phantom-firmware-tools/wiki/Firmware-m0800#debug-serial-interface-368_u0

-------------------------------
Loading from nand0, offset 0x900000
   Image Name:   Linux-2.6.32.17-davinci1
   Created:      2015-03-30   9:54:39 UTC
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    4032832 Bytes = 3.8 MiB
   Load Address: 80008000
   Entry Point:  80008000
## Booting kernel from Legacy Image at 80700000 ...
   Image Name:   Linux-2.6.32.17-davinci1
   Created:      2015-03-30   9:54:39 UTC
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    4032832 Bytes = 3.8 MiB
   Load Address: 80008000
   Entry Point:  80008000
   Loading Kernel Image ... OK
------------------------------

If the uImage is corrupted, what is this image that it's loading? I can send commands to linux through the 368_U0 serial interface and it will execute them (I can list the processes and also kill them). Also, it appears to do it's job, apart from updating. This got me thinking that maybe your uImage is intact too (and at the same address), and only the command to load it got somehow corrupted. What do you think? If that is the case, maybe you could try launching it manually, from a uboot prompt. Also, if what I said 'till now is completely wrong, please disregard it, because I litterally don't know anything about Linux, apart from what I read these last two days.

Getting back at my issue, what else do you think could cause my "no updating" symptom? What other diagnostics can I run?

**Comment by MrBurnsAT** [comment]
Do u know which firmware is in ur Bird?

If yes. I would try to Update m0800 and m1500 with 
PDM368Fw3.bin (m0800) and P68013Fw3.bin (m1500) from correct firmware on SD Card

**Comment by DannyBoyCJ** [comment]
Truth is, I'm not quite sure. The last fully flashed firmware was "Phantom_3_Professional_Firmware_v1.2.6_en", which ended in failure. Then I attempted to flash "Phantom_3_Professional_Firmware_v1.1.9_en", but that process didn't finish, and broke my drone. So, maybe it's 1.2.6, or maybe a combination with 1.1.9.... I should probably extract them from version 1.2.6.
I never used the tools from this repository before, so I guess I'm about to learn some new skills :))

**Comment by MrBurnsAT** [comment]
If u want i can extract it for u and Post it here.

Wich Update did u try?
1.1.9 and 1.2.6?

Try 1.3.2 or higher with debug File!

**Comment by MrBurnsAT** [comment]
PS bevore u van Downgrade from 1.2.6 down to 1.1.9 u must use MVOM0Fw.bin 
After that u can go down

**Comment by DannyBoyCJ** [comment]
That was probably my mistake, I tried going to 1.1.9 without using MVOM0Fw.bin first. I tried it now, but it doesn't work, either. If you could extract those files for me it would really help me.  I tried most of the versions until now, and none worked. It would be useful if I could "listen in" on the updating process through the serial connection, but I don't know which is the best interface for monitoring this, or if it is even possible. U0 and 368_U0 do not give signs of updating. On the U0 I can see an updating log, but only when the camera is disconnected from the main gimbal board, and it is a very short process. It looks like this:

[putty_05_07_2017_162816.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1135563/putty_05_07_2017_162816.txt)

So I will try to see if it works with the debug file now on version 1.3.2, as you suggested, and if it doesn't, I'll give it a try with the extracted firmware for m0800 and m1500 (probably from 1.2.6).

**Comment by DannyBoyCJ** [comment]
Nope, firmware + debug doesn't work, and neither does  MVOM0Fw.bin. The drone acts like there's no firmware on card.

**Comment by MrBurnsAT** [comment]
Why did u try an Update to V01.09.0060???

Ans Not 1.3.2???

Dont do so big Version jumps!

Now i Need the log File from sd Card After Update.

/misc/log/P3X_FW_LOG_AB.txt

**Comment by DannyBoyCJ** [comment]
I didn't try 1.09.0060, but "Phantom_3_Professional_Firmware_v1.1.9_en", the one before 1.2.6. The last log my drone wrote is from 6 weeks ago, when it got screwed up during downgrading. It hasn't written a log since. This is it, the log from 6 weeks ago:
[P3X_FW_LOG_AB.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1136418/P3X_FW_LOG_AB.txt)

**Comment by MrBurnsAT** [comment]
Ur Putty log says something different



[0m[00000820][CA9_0] Packet Detected / c:\P3X_FW_V01.09.0060.bin /.

**Comment by DannyBoyCJ** [comment]
You are right, I probably wanted to use 1.1.9 and used 1.09 instead... Anyway it does exactly the same thing no matter what version of firmware is on the card, but only if the camera is disconnected from the gimbal main board. If the drone is fully assembled, the serial log looks like this:

[putty_10_07_2017_U0.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1136637/putty_10_07_2017_U0.txt)

It's like it never detects there's a firmware file on the SD card. So, my guess is that when it can't detect the camera, it tries to update, because it thinks it's a hardware error. But when the camera is present, it has no reason to look for an upgrade. It would be useful to know which process is responsible for starting the upgrade process. Maybe it can be initiated manually, from the linux prompt. For reference, here is a list of active processes, displayed with the "top" command:

[7m  PID  PPID USER     STAT   VSZ %VSZ CPU %CPU COMMAND[0m
  428     1 root     S <  83256185.9   0 61.4 /opt/dji/bin/encode_usb -a -s
  437   431 root     R     3140  7.0   0  7.6 top
  431   430 root     S     3144  7.0   0  0.0 -sh
    1     0 root     S     3140  7.0   0  0.0 init
  430     1 root     S     3140  7.0   0  0.0 {autologin} /bin/sh /sbin/autologi
  427     1 root     S     1748  3.9   0  0.0 /opt/dji/bin/monitor
    5     2 root     SW       0  0.0   0  0.0 [khelper]
    4     2 root     SW       0  0.0   0  0.0 [events/0]
    3     2 root     SW       0  0.0   0  0.0 [ksoftirqd/0]
    2     0 root     SW       0  0.0   0  0.0 [kthreadd]
    8     2 root     SW       0  0.0   0  0.0 [async/mgr]
  137     2 root     SW       0  0.0   0  0.0 [sync_supers]
  139     2 root     SW       0  0.0   0  0.0 [bdi-default]
  141     2 root     SW       0  0.0   0  0.0 [kblockd/0]
  155     2 root     SW       0  0.0   0  0.0 [khubd]
  158     2 root     SW       0  0.0   0  0.0 [kseriod]
  190     2 root     SW       0  0.0   0  0.0 [rpciod/0]
  199     2 root     SW       0  0.0   0  0.0 [kswapd0]
  200     2 root     SW       0  0.0   0  0.0 [aio/0]
  201     2 root     SW       0  0.0   0  0.0 [nfsiod]

**Comment by MrBurnsAT** [comment]
Here are the files from 1.02.0006

i added it 2 times! 

2 files with old filename (without P at start, and without 3 at end)
2 files with new filename (with P at start, and with 3 at end)

[1.02.0006.zip](https://github.com/mefistotelis/phantom-firmware-tools/files/1136869/1.02.0006.zip)

**Comment by DannyBoyCJ** [comment]
Thank you! I'll give them a run, see what happens!

**Comment by DannyBoyCJ** [comment]
Ok, so good news so far. I copied the files with P at start, and with 3 at end on the SD card, and my bird started updating. And it finished succesfully! However, it still won't take the normal 1.2.6 bin placed on SD. So I'm thinking: can we do the same thing for all the modules? Maybe flashing all of them this way will finally put a consistent (and fully working) firmware on the drone. Here is the log it wrote now:
[P3X_FW_LOG_AB.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1137173/P3X_FW_LOG_AB.txt)
The way I see it from the log file, last time it didn't finish upgrading module 0100. So maybe that's where my problem is. Is m0100 upgradeable in this way? Anyway, very good work, MrBurnsAT !!

**Comment by MrBurnsAT** [comment]
pls here is m0100

[PFC300XFw3.zip](https://github.com/mefistotelis/phantom-firmware-tools/files/1137222/PFC300XFw3.zip)

**Comment by DannyBoyCJ** [comment]
Awesome, let's see how this goes :)

**Comment by DannyBoyCJ** [comment]
Ok, so it looks like it starts updating, but it actually doesn't. This is the log:
[P3X_FW_LOG_AB.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1137251/P3X_FW_LOG_AB.txt)
And this is the log from Ambarella console (U0 pads):
[putty_11_07_2017_PFC300XFw3.txt](https://github.com/mefistotelis/phantom-firmware-tools/files/1137260/putty_11_07_2017_PFC300XFw3.txt)
What I find strange is that although the putty log states that it can't find firmware on the SD, the AB log shows that it is trying to program m0100, but fails with the same error as last time. There's something obviously wrong with module 0100. From what I see, modules 0100 and 0101 are part of Ambarella firmware. Hmm... Should I try this with the files for all the modules on the SD? I know that they don't work if they are encrypted, so... can the encrypted ones be decrypted and prepared for flashing?

**Comment by MrBurnsAT** [comment]
No they dont Need to be decrypted!

Only rename to right Name.
And put on sd

**Comment by MrBurnsAT** [comment]
here is the whole 1.2.0006 with all modules extracted

with little manual info.txt and filenames.txt

https://www.dropbox.com/s/dtgrz2bb2l4rzcq/P3X_FW_V01.02.0006_extracted.zip?dl=0

**Comment by DannyBoyCJ** [comment]
Ok, thank you very much, I'll try the whole package next :)

**Comment by DannyBoyCJ** [comment]
Apparently, upgrading only fails when the upgrade for m0100 (PFC300XFw3.bin) is detected. Is there another way to flash this particular package?

**Comment by DannyBoyCJ** [comment]
Update, while trying to update m0100, and "listening in" on the Ambarella U0 pads, it reported "ram fail":

>--- ARM DATA ABORT EXCEPTION ---ram fail.
>Oops: CPU Exception!
pc : [<00011324>]    lr : [<00000000>]
sp : 0006f328  ip : 00000000  fp : 00000000
r10: 0006b400  r9 : 00000000  r8 : 00000000
r7 : 00000000  r6 : 00000000  r5 : 00000000
r4 : 0006f394  r3 : 00000000  r2 : 00000000
r1 : fff90c5b  r0 : ffffffa4
Flags: NzCv
IRQs on  FIQs on  Mode SVC_32

I might have another issue on my hands here...

**Comment by mefistotelis** [comment]
Here you can check which chips are bound to which:
https://github.com/mefistotelis/phantom-firmware-tools/wiki/WM320-Gimbal-top-board#programming

I updated the Ambarella firmware page with all I know; when you'll learn the other methods please describe them so that I can update:
https://github.com/mefistotelis/phantom-firmware-tools/wiki/Firmware-m0100

The "other methods" may be either something listed in Ambarella SDK, or a way of flashing the NAND chip directly.

**Comment by DannyBoyCJ** [comment]
Thanks, I'll write here as soon as I got something.

**Comment by jjhdtv** [comment]
Looks like this thread is a few months old, however I have started a journey down a very similar path.
Using MrBurnsAT tools and ubl1.img , u-boot.img files I have successfully restored my nand chip.  I used the following command (sfh_DM36x.exe -nandflash -v -p "COM3" ubl1.img u-boot.img).  

Boot log when not attached to any other device gets to a point, throws a few errors and continuous reboots itself repeating the same sequence.  Below are the last few lines from putty just prior to the rebooting.  I'm at a lost on what I should try next.  Like DannBoyCJ problem, my issue started with getting a failed firmware update attempt.  Log showed the infamous 08/15 error.  Appreciate any help or suggestions.  thanks.   (putty log portion  below)
............
root@Dji-Pro:~#
encode_usb V00.02.02.00(Jul 18 2015 16:52:39).
Entry Encrypt qury mode
init  libusb successed!!!
!!!!!!!!!!!!!!!![    6.720000] EVM: switch to ambarella video input
!!!!!!! open 68013 failed   VENDOR_ID = 0x0547
Init libusb  faild!
Auto to detect video input type:
Default video input selected:Hdmi_yuv
S[    6.730000] davinci_resizer davinci_resizer.2: RSZ_G_CONFIG:0:1:124
etting video standard to: 0x1000[    6.740000] davinci_previewer davinci_previewer.2: ipipe_set_preview_config
00000000000
Cap[    6.750000] vpfe-capture vpfe-capture: IPIPE Chained
ture input set t[    6.760000] vpfe-capture vpfe-capture: Resizer present
o Hdmi_yuv:11
Capture videoStd[    6.770000] EVM: switch to ambarella video input
Old=0000000b
[    6.770000] vpfe-capture vpfe-capture: width = 1280, height = 720, bpp = 1
[    6.780000] vpfe-capture vpfe-capture: adjusted width = 1280, height = 720, bpp = 1, bytesperline = 1280, sizeimage = 1382400
[    6.790000] vpfe-capture vpfe-capture: width = 1280, height = 720, bpp = 1
[    6.800000] vpfe-capture vpfe-capture: adjusted width = 1280, height = 720, bpp = 1, bytesperline = 1280, sizeimage = 1382400
videoFrameRate=30000, width=1280, height=720
Venc1.c : minNumOutBufs 2, minOutBufSize[0] 1382400, minOutBufSize[1] 800
GOP interval:30, outBufSize:1382400, enAdpBW:1, airRate=0, slice=6, VUI=4,timeScale=60001
osdThrFxn: process ID is 433 ***********
watchdog timeout:5
ctrlThrFxn: process ID is 425 ***********
Frmae:1 Pslice = 0, Islice = 8[44ms]

GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1

usblink status check~~~~~~~~~~~~~~~~
GetFPGAstatus err!!
inptVD=0547
USB device go down, need to reboot system
usb bus error dectected!
No USB device ID:547
try reset 68013....!
umount: can't remount none read-only
umount: mdev busy - remounted read-only
The system is going down NOW!
Sent SIGTERM to all processes
Sent SIGKILL to all processes
Requesting system reboot
[   10.470000] Restarting system.

DM36x initialization passed!
UBL Product Vesion : DJI-NOHDMI-WM610-UBL-1.0-rc0(2015-01-13)
Dji UBL Version: 1.51(Jan 13 2015 - 16:42:11)
Booting Catalog Boot Loader
BootMode = NAND
Starting NAND Copy...
Valid magicnum, 0xA1ACED66, found in block 0x00000019.
   DONE
Jumping to entry point at 0x81080000.

U-Boot Product Vesion : DJI-WM610-Uboot-1.0-rc4(2014-08-13)
U-Boot 2010.12-rc2-svn-Dji (Aug 13 2014 - 10:55:23)
Cores: ARM 297 MHz
DDR:   270 MHz
I2C:   ready
DRAM:  128 MiB
NAND:  128 MiB
Bad block table found at page 65472, version 0x01
Bad block table found at page 65408, version 0x01
*** Warning - bad CRC, using default environment

Net:   Ethernet PHY: GENERIC @ 0xff
DaVinci-EMAC
Press ESC to abort autoboot in 1 seconds

Loading from nand0, offset 0x4a0000
** Unknown image type
Wrong Image Format for bootm command
ERROR: can't get kernel image!

Loading from nand0, offset 0x900000
   Image Name:   Linux-2.6.32.17-davinci1
   Created:      2015-03-30   9:54:39 UTC
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    4032832 Bytes = 3.8 MiB
   Load Address: 80008000
   Entry Point:  80008000
## Booting kernel from Legacy Image at 80700000 ...
   Image Name:   Linux-2.6.32.17-davinci1
   Created:      2015-03-30   9:54:39 UTC
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    4032832 Bytes = 3.8 MiB
   Load Address: 80008000
   Entry Point:  80008000
   Loading Kernel Image ... OK
OK

**Comment by mefistotelis** [comment]
I don't see any error messages besides those caused by the board not being connected to OFDM.

Have you tried just connecting it?

**Comment by jjhdtv** [comment]
Not yet.  I have the gimbal all apart.  You think I could plug in the
external connectors (leave the camera disconnected) and power with the 3.3v
connected to the mainboard pins?  or some other temporary setup.

I was curious about the usb error and the "open 68013 failed   VENDOR_ID =
0x0547"

**Comment by mefistotelis** [comment]
> You think I could plug in the external connectors (leave the camera disconnected) and power with the 3.3v

As long as ground voltages are leveled, everything will work.

> I was curious about the usb error and the "open 68013 failed   VENDOR_ID = 0x0547"

Davinci is connected to CY68013 chip (on OFDM board) via USB interface. CY68013 is an USB controller, after all.

To get an understanding how boards are connected, you can study the simplified schematics:
https://github.com/mefistotelis/dji-hardware-schematics/blob/master/phantom_3_pro_overview/phantom_3_pro_overview.pdf

**Comment by jjhdtv** [comment]
Ok - im back at it.  Let me explain my progress (or in this case lack of progress)
I connected the main gimbal board (without the motors and camera attached) to the 2 connectors on the p3p frame.  I powered up the quad with its own battery and left my ftdi attached as before (u368 pins).

Powered up the p3p  and made a terminal connection in Putty.  Gimbal board led green was blinking but no data was transmitted to my ftdi - putty screen blank - ftdi no data flow lights blinking.

I disconnected the main board from the ofdm and esc board connector, applied 3.3v to pads, and started a terminal connection.  As before I got the boot log displayed above in my previous post.

I also tried updating the firmware by  the sd card method (P3X_FW_V01.11.0020.bin)  while the mainboard was attached to the p3p as described above.   It did not work.  I'm assuming if the camera is not connected - the sd card is not read but not sure.

So, I'm a bit confused on how to proceed.  Should I bite the bullet and assemble everything (camera, gimbal motors, mainboard, external plugs to ofdm and esc board)?  and hope everything boots ok.  Or is there a way of further testing firmware and hardware without putting everything back together?

Not sure why not RX TX data is captured when only the mainboard is connected to the OFDM and ESC boards.  Do I need to connect to the other RX TX pins on the board?

**Comment by jjhdtv** [comment]
correction to above

Not sure why not RX TX data is **not** captured when only the mainboard is connected to the OFDM and ESC boards. Do I need to connect to the other RX TX pins on the board?

**Comment by mefistotelis** [comment]
Remember you need RX TX and GND for FTDI.

If DM365 gives no sign of life, maybe it's the most usual problem - power not connected. Check voltages on the board.

**Comment by jjhdtv** [comment]
Success!!
I plugged everything back in (camera, ofdm, esc).  Reconnected ground, pwr, rx,tx to the ftdi.  Powered the p3p with battery - opened Putty connection.  All appears good.  Connected with the RC, received the live video image.  See boot log below.   Now does anyone know how to prevent the nand corruption in the future?  Access to reflash is a pain.  Thanks for the assistance.  


Filesystem Product Vesion : DJI-WM610-Filesystem-1.0-rc3(2014-06-06)
Runing /etc/init.d/rcS
[    4.060000] CMEMK module: built on Jan 13 2015 at 19:24:09
[    4.060000]   Reference Linux version 2.6.32
[    4.080000]   File /home/DJI2/Project/enc_nohdmi_wm610/src/dvsdk/dvsdk_dm368_4_02_00_06/linuxutils_2_26_01_02/packages/ti/sdo/linuxutils/cmem/src/module/cmemk.c
[    4.100000] allocated heap buffer 0xc7000000 of size 0x2456000
[    4.130000] heap fallback enabled - will try heap if pool buffer is not available
[    4.130000] CMEM Range Overlaps Kernel Physical - allowing overlap
[    4.150000] CMEM phys_start (0x1000) overlaps kernel (0x80000000 -> 0x83000000)
[    4.160000] cmemk initialized
[    4.200000] IRQK module: built on Jan 13 2015 at 19:23:41
[    4.200000]   Reference Linux version 2.6.32
[    4.220000]   File /home/DJI2/Project/enc_nohdmi_wm610/src/dvsdk/dvsdk_dm368_4_02_00_06/linuxutils_2_26_01_02/packages/ti/sdo/linuxutils/irq/src/module/irqk.c
[    4.240000] irqk initialized
[    4.300000] EDMAK module: built on Jan 13 2015 at 19:23:04
[    4.320000]   Reference Linux version 2.6.32
[    4.320000]   File /home/DJI2/Project/enc_nohdmi_wm610/src/dvsdk/dvsdk_dm368_4_02_00_06/linuxutils_2_26_01_02/packages/ti/sdo/linuxutils/edma/src/module/edmak.c
sulogin: no password entry for root
login[437]: root login on 'ttyS0'
#Process /etc/profile...
root@Dji-Pro:~#
encode_usb V00.02.02.00(Jul 18 2015 16:52:39).
Entry Encrypt qury mode
init  libusb successed!!!
open 68013 succe[    6.790000] EVM: switch to ambarella video input
ed  VENDOR_ID = 0x0547
Claiming libusb_claim_interface 68013 successed
Auto to detect video input type:
Default video input selected:Hdmi_yu[    6.810000] davinci_resizer davinci_resizer.2: RSZ_G_CONFIG:0:1:124
v
Setting video standard to: 0x[    6.820000] davinci_previewer davinci_previewer.2: ipipe_set_preview_config
[    6.820000] vpfe-capture vpfe-capture: IPIPE Chained

Capture input s[    6.830000] vpfe-capture vpfe-capture: Resizer present
et to Hdmi_yuv:11
Capture videoStd[    6.840000] EVM: switch to ambarella video input
Old=0000000b
[    6.840000] vpfe-capture vpfe-capture: width = 1280, height = 720, bpp = 1
[    6.850000] vpfe-capture vpfe-capture: adjusted width = 1280, height = 720, bpp = 1, bytesperline = 1280, sizeimage = 1382400
[    6.860000] vpfe-capture vpfe-capture: width = 1280, height = 720, bpp = 1
[    6.870000] vpfe-capture vpfe-capture: adjusted width = 1280, height = 720, bpp = 1, bytesperline = 1280, sizeimage = 1382400
videoFrameRate=30000, width=1280, height=720
Venc1.c : minNumOutBufs 2, minOutBufSize[0] 1382400, minOutBufSize[1] 800
GOP interval:30, outBufSize:1382400, enAdpBW:1, airRate=0, slice=6, VUI=4,timeScale=60001
osdThrFxn: process ID is 442 ***********
watchdog timeout:5
ctrlThrFxn: process ID is 434 ***********
Frmae:1 Pslice = 0, Islice = 8[44ms]
GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1
ARM Load: 61% Video fps: 7 fps Video bit rate: 0 kbps PHY Bandwidth: 0 kbps Time: 00:00:01 Resolution: 1280x720

GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1
ARM Load: 4% Video fps: 8 fps Video bit rate: 0 kbps PHY Bandwidth: 0 kbps Time: 00:00:02 Resolution: 1280x720

GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1
ARM Load: 6% Video fps: 7 fps Video bit rate: 0 kbps PHY Bandwidth: 0 kbps Time: 00:00:03 Resolution: 1280x720

GetFPGAstatus err!
FPGA[1,0,1,0]-1.000000:0
hdmi status is 1
ARM Load: 3% Video fps: 8 fps Video bit rate: 0 kbps PHY Bandwidth: 0 kbps Time: 00:00:04 Resolution: 1280x720

Frmae:32 Pslice = 0, Islice = 8[32ms]
FPGA[40,100,1,0]-1.000000:4000000
hdmi status is 1
ARM Load: 7% Video fps: 8 fps Video bit rate: 564 kbps PHY Bandwidth: 4000 kbps Time: 00:00:05 Resolution: 1280x720

Frmae:41 Pslice = 0, Islice = 8[28ms]
Encrypt passed!
FPGA[80,100,1,0]-1.000000:8000000
hdmi status is 1
ARM Load: 16% Video fps: 18 fps Video bit rate: 2643 kbps PHY Bandwidth: 8000 kbps Time: 00:00:06 Resolution: 1280x720

FPGA[80,100,1,0]-1.000000:8000000
hdmi status is 1
ARM Load: 34% Video fps: 30 fps Video bit rate: 5053 kbps PHY Bandwidth: 8000 kbps Time: 00:00:07 Resolution: 1280x720

**Comment by mefistotelis** [comment]
>  does anyone know how to prevent the nand corruption in the future? 

Two simple commandments:
1. You shall not upgrade firmware unless you have genuine issue with current one.
2. You shall not turn on your bird without flying, as this overheats the gimbal top board.

**Comment by MrBurnsAT** [comment]
One commandment if me.

If u dont use ur Phantom for long time.
Turn on ur Phantom once every Month for at least 2 Minutes. That the DM368 firmware can find new bad blocks and exclude them.

**Comment by jjhdtv** [comment]
Thanks again for all of the good suggestions.  Video transmission is still working.  Unrelated, issue the gimbal yaw motor is behaving erratically.  Everything appears to boot fine, but then after initializing the gimbal starts to turn side to side max distance (about 10 to 20 seconds after initialization).  Tried all of the calibrations (imu, gimbal).  Did not fix.  

I'm guessing something went wrong with my reassembly.  Any Ideas?  Set screw is solid on the flat part of the shaft.  Seems more like a sensor or motor issue.

**Comment by mefistotelis** [comment]
Hard to tell. Any of the gimbal arms could cause that, by either having circuit issue, potentiometer issue or flat wire connection issue.

**Comment by jjhdtv** [comment]
Only touched the yaw gimbal motor on the mainboard.  All over cables and motors left intact.   Gimbal has never been in a crash.  I'll start at the mainboard and yaw motor.

**Comment by chainwood008** [comment]
My P3Pro Camera mode is clean but when change to video mode showing the purple color for a few second and clean again. this purple color several time showing on video mode.
pls help me...
Thanks You

**Comment by DannyBoyCJ** [comment]
Ok, I am back at it. I tried to flash all modules of the P3X_FW_V01.03.0020 separately, leaving out m0100, the one that didn't want to update.  I got errors on m0305 and m0306, and their corresponding "bin" files have been renamed to "PMCAPPFw3_damaged.bin" and "PMCLDRFw3_damaged.bin" during the update process. The gimbal is now unresponsive and doesn't go through the initial checks when starting. I believe this happened because of the modules being encrypted, I forgot to check for that before flashing. So... is there a way to flash those modules with the unencrypted versions, other then by official firmware on sd? Or is it really dead this time...

**Comment by mefistotelis** [comment]
> I got errors on m0305 and m0306, and their corresponding "bin" files have been renamed to "PMCAPPFw3_damaged.bin" and "PMCLDRFw3_damaged.bin" during the update process.

And is the flight controller working? Can you start the motors?
Non-working FC can be spotted by motors starting to beep forever soon after startup.

> The gimbal is now unresponsive and doesn't go through the initial checks when starting. I believe this happened because of the modules being encrypted

It is possible, m0400 is usually encrypted.

> So... is there a way to flash those modules with the unencrypted versions, other then by official firmware on sd?

If you want to write m0400, it should be possible to use ST-link, as in all STM micro-controllers. You'll have to figure out what should be connected where though, I never tried to do this.

**Comment by DannyBoyCJ** [comment]
I can't start the motors, but then again, neither could I start them before this. The remote has very limited connectivity and I can't issue any commands, it only gets some info from the drone, possibly because the firmware on the remote is now so far away (1.10) from the one on the bird (1.03). The motors don't beep at all.
However, the log reports the m0400 upgrade as successful:

```
 [00366526]========== remo-con disconnect. boot(15) ============
 [00366603]Firmware [C:\PGIMBALFw3.bin] detected, card sn [0x28537f3c].
 [00366686][04 00] Firmware upgrade start...
 [00424013][04 00] Firmware upgrade finished successfully.
 [00424097]Done.
```

The only failed modules are m0305 and m0306:

```
[00524453]========== remo-con disconnect. boot(15) ============
 [00524530]Firmware [C:\PMCLDRFw3.bin] detected, card sn [0x28537f3c].
 [00524612][03 05] Firmware upgrade start...
 [00553203][03 05] Firmware upgrade finish failed (step = 4, err = 0xf7, idx = 29, los = 0, ng = 30).
 [00583295]Done.

 [00038889]========== remo-con disconnect. boot(15) ============
 [00038966]Firmware [C:\PMCAPPFw3.bin] detected, card sn [0x28537f3c].
 [00039049][03 06] Firmware upgrade start...
 [00081627][03 06] Firmware upgrade finish failed (step = 2, err = 0xaa).
 [00111712]Done.
```
So, could the gimbal unresponsiveness be due to firmware inconsistency, then? Because the DJI app now reports it that way. I'm attaching the whole log from the update here for reference. The firmware version I am trying to use is P3X_FW_V01.03.0020 (one step up from the one I think is on the drone).

[P3X_FW_LOG_AB.txt](https://github.com/o-gs/dji-firmware-tools/files/1800506/P3X_FW_LOG_AB.txt)

I later on discovered that you had provided an updated version of dji_fwcon.py in the tools section, with decryption capabilities, and used that to extract m0305 an m0306. They still fail updating, as visible in the log above.

> If you want to write m0400, it should be possible to use ST-link, as in all STM micro-controllers. You'll have to figure out what should be connected where though, I never tried to do this.

I wasn't referring to m0400, but to m0305 and m0306, because they were the only unsuccessful ones.

**Comment by garykrogman** [comment]
I have repaired a few P3P cameras using the methods on this page. Thank you all for your help! I also have a inspire 1. I have a X3 and an X5 camera. X3 works fine so the problem is not the aircraft or rc. The X5 has the same symptoms as the P3P gimbal you guys have figured out how to repair. 08 00 and 15 00 device not detected. Gimbal functions and video is written to the sd properly. Before I crack open the X5 does anybody have any experience with the X5?

**Comment by mefistotelis** [comment]
Inspire 1 contains the same circuit as Ph3 Pro, just fitted on different boards.

If you have any good quality camera, you may help us by taking photos of each board within your X5. We currently don't have photos which would allow to identify all components. Remember to do both sides of each board.

**Comment by garykrogman** [comment]
Here are a few pictures of the x5. I will be sending more. I have 2 that I
will not be able to put back together until I find a source for the yaw
motors. If you want I can ship you one. Both have the same symptom, able to
record to the sd but no video stream.
      Gary

**Comment by mefistotelis** [comment]
You did not shared any pictures...

**Comment by garykrogman** [comment]
![img_1727](https://user-images.githubusercontent.com/29103635/50035294-41ad7500-ffcf-11e8-8eaa-d191cbb10f86.jpg)

**Comment by garykrogman** [comment]
I now have better photos of The Zenmuse x5 video processor board. I am trying to what test points to connect my serial interface to. I would think it would be U0_TX and RX. Does anyone know if it is the same firmware package as the NAND on the p3p? Then there would be b+ and boot configuration. 
![img_1747](https://user-images.githubusercontent.com/29103635/50059506-ccb87780-0155-11e9-9544-6bb06e7e476d.jpg)
![img_1748](https://user-images.githubusercontent.com/29103635/50059512-d4781c00-0155-11e9-9f59-e556c281bfbd.jpg)

**Comment by mefistotelis** [comment]
Thanks for the photos!

You should be able to extract DM36x firmware, and further extract boot loader from it, using the Inspire FW package. There is no reason to risk using P3X firmware.

Also, this board does NOT contain DM3xx.. this is not the one you should program.
There's only Ambarella, its NAND (Micron chip) and its RAM (Elpida chip). Plus gimbal control (STM32).
The rest is just power supply for these chips.

**Comment by garykrogman** [comment]
You were correct! It was on the tilt gimbal side. I have been in computers and electronics all my life, but extracting firmware and a boot loader from a firmware package, I need help with. Here is the board with the CMOS and the one with the DM368.
![img_1753](https://user-images.githubusercontent.com/29103635/50060695-283e3180-0165-11e9-966b-a0ce5d00142e.jpg)
![img_1755](https://user-images.githubusercontent.com/29103635/50060697-2d02e580-0165-11e9-8e55-7c096b64b532.jpg)

**Comment by garykrogman** [comment]
I have the TI utility. I believe I need to extract m0800 and m1500 bin files from the full bin file, then repackage them and apply it to the DM368. I s this the correct plan?

**Comment by mefistotelis** [comment]
> I believe I need to extract m0800 and m1500 bin files from the full bin file, then repackage them and apply it to the DM368. I s this the correct plan?

No, no need to re-package.

1. You only need to fix m0800. The Cypress chip (m1500) is missing only because communication to it is forwarded through m0800.

2. You only need to extract the FW. The usual cause for this error is that bootloader got damaged on NAND which contains programming for DM3xx. You only need to extract the bootloader.

3. When you have the bootloader files, you just have to flash them via serial. See details here:
https://github.com/o-gs/dji-firmware-tools/wiki/Flashing-firmware-on-DaVinci-media-processors

Note that there is a chance that the issue is not with bootloader - ie. another part of DaVinci Linux needs to be reflashed, or the NAND chip is permanently damaged and needs replacing. But in most cases reflashing bootloader works.

**Comment by garykrogman** [comment]
Yes, I will take a shot of the other side. I have made some progress. I user python to extract all the module files. What do I use to extract the bootloaders from the m0080.bin file?

**Comment by mefistotelis** [comment]
> What do I use to extract the bootloaders from the m0080.bin file?

https://github.com/o-gs/dji-firmware-tools/wiki/Firmware-m0800#structure

**Comment by garykrogman** [comment]
I now have img files extracted from m0800 bin. The processor speed is 432. I notice there are 2 432 img files with the same size. Should they be combined some how, or what one do I use?
![capture](https://user-images.githubusercontent.com/29103635/50376558-547e0600-05dc-11e9-9b84-6eb4b3fb6d3d.JPG)

2nd side of the DM368 board
![img_1769](https://user-images.githubusercontent.com/29103635/50376623-64e2b080-05dd-11e9-8bb4-056e0477fcd7.jpg)

**Comment by mefistotelis** [comment]
>  I notice there are 2 432 img files with the same size. Should they be combined some how, or what one do I use?

It doesn't matter, they're identical. Just remember to remove their headers as the 'structure' section explains.
